time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java

```diff
@@ -288,11 +340,40 @@ private Duration checkRegularSpacing() {
                     StringDataChunk chunk = new UncompressedStringDataChunk(0, stringValues.toArray(new String[0])).tryToCompress();
                     timeSeriesList.add(new StringTimeSeries(metadata, chunk));
                 } else {
-                    throw assertDataType(dataTypes[i - 2]);
+                    throw assertDataType(dataTypes[i - fixedColumns]);
                 }
             }
             return timeSeriesList;
         }
+
+        private TimeSeriesIndex getTimeSeriesIndex() {
+            Long spacing = checkRegularSpacing();
+            if (spacing != Long.MIN_VALUE) {
+                return new RegularTimeSeriesIndex(times.get(0), times.get(times.size() - 1), spacing);
+            } else {
+                return new IrregularTimeSeriesIndex(times.stream().mapToLong(l -> l).toArray());
+            }
+        }
+
+        private long checkRegularSpacing() {
+            if (times.size() < 2) {
+                throw new TimeSeriesException("At least 2 rows are expected");
+            }
```

Comment:
Is an empty timeseries totally dumb? Maybe it could be useful for the timeline (see my comment about null TS vs empty TS in #1539)

Reply:
At this point it only affects the reading of TimeSeries from a CSV file, it is possible to create an empty object of type TimeSeries without using a CSV file as a starting point